### PR TITLE
Adding task list in device info message

### PR DESCRIPTION
**Pull request by zed-rishabh**
We are adding a list of tasks present on the device in the device info message so that the tests can verify the status of the task after deleting the app instance.

Tests can also it while device clean-up to make sure the device has no active tasks.

cc: @kalyan-nidumolu

### CODE REVIEW

File: api/proto/info/info.proto
Comment (kalyan-nidumolu) [inline api/proto/info/info.proto:253]: Are there different kinds of tasks? Or is there only one? Just wondering if there is a need for "type" field.

**Review by kalyan-nidumolu — APPROVED** [review]
LGTM

File: api/proto/info/info.proto
Comment (rvs) [inline api/proto/info/info.proto:250]: Hey @zed-rishabh -- this definitely requires more documentation in reflection in our API doc before this could be merged. For example, what is this ID, what is the name. Also, I really don't think status should be a string -- it probably should map to the other probof sturct.

**Review by rvs — CHANGES_REQUESTED** [review]

File: api/proto/info/info.proto
Comment (zed-rishabh) [reply api/proto/info/info.proto:253]: >Are there different kinds of tasks?

No

File: api/proto/info/info.proto
Comment (zed-rishabh) [reply api/proto/info/info.proto:250]: I added it in api v2 docs.

Comment (rvs) [reply api/proto/info/info.proto:250]: Cool -- so this is representation of containerd tasks -- thanks for making this clear in the docs, but I have one more question now and I'm sure @deitch will have an opinion or two ;-)

File: api/proto/info/info.proto
Comment (rvs) [inline api/proto/info/info.proto:252]: What's the difference between ID and name?

**Review by rvs — CHANGES_REQUESTED** [review]
Lets clarify id vs. name (and document it in the proto file) and then this can be merged

File: api/proto/info/info.proto
Comment (zed-rishabh) [reply api/proto/info/info.proto:252]: What do you want me to add in the proto file? I thought it is self-explanatory that id is representing the task's unique id and the name is representing the task name in the containerd world. I am not sure what can I add to the proto file. Can you please help me in understanding the requirement here?

Comment (rvs) [reply api/proto/info/info.proto:252]: That's literally my question -- what is that unique ID? I don't understand it since it doesn't seem to be a containerd concept, is it @zed-rishabh ?

**Review by rvs — APPROVED** [review]
LGTM

**Comment by rvs** [comment]
So... @zed-rishabh -- I'm confused as to what needs to be done with this PR -- did it go in as part of another PR?

At any rate -- I'll let you decide and either close it or rebase it so we can merge

**Comment by zed-rishabh** [comment]
> So... @zed-rishabh -- I'm confused as to what needs to be done with this PR -- did it go in as part of another PR?
> 
> At any rate -- I'll let you decide and either close it or rebase it so we can merge

Conflicts resolved. We can merge it now.

**Pull request merged**
